data/data.rb

```diff
@@ -0,0 +1,169 @@
+        book = Book.new(book['title'], book['author'])
+        @books << book
+      end
+    end
+    file.close
+  end
+end
+
+def save_book(title, author)
+  obj = {
+    title: title,
+    author: author
+  }
+
+  if File.exist?('./data/books.json')
+    file = File.open('./data/books.json')
+
+    if File.empty?('./data/books.json')
+      book = [obj]
+    else
+      book = JSON.parse(File.read('./data/books.json'))
+      book << obj
+    end
+
+    file.close
+
+    myfile = File.open('./data/books.json', 'w')
+    myfile.write(JSON.pretty_generate(book))
+    myfile.close
+  end
+end
+
+def load_people
+  if File.exist?('./data/people.json')
+    file = File.open('./data/people.json')
+    if File.empty?('./data/people.json')
+      puts 'Please add people data if this is your first time visiting our app'
+    else
+      people = JSON.parse(File.read('./data/people.json'))
+      people.each do |person|
+        if person['option'] == 'Student'
+          student = Student.new(person['age'], person['name'])
+          @people << student
+        else
+          teacher = Teacher.new(person['specialization'], person['age'], person['name'])
+          @people << teacher
+        end
+      end
+    end
+    file.close
+  else
+    puts 'Please insert some data'
+  end
+end
+
+def save_student(age, name, parent_permission)
+  obj = {
+    type: 'Student',
+    age: age,
+    name: name,
+    parent_permission: parent_permission
+  }
+
+  if File.exist?('./data/people.json')
+    file = File.open('./data/people.json')
+
+    if File.empty?('./data/people.json')
+      student = [obj]
+    else
+      student = JSON.parse(File.read('./data/people.json'))
+      student << obj
+    end
+
+    file.close
+
+    myfile = File.open('./data/people.json', 'w')
+    myfile.write(JSON.pretty_generate(student))
+    myfile.close
+  end
+end
+
+def save_teacher(specialization, age, name)
+  obj = {
+    type: 'Teacher',
+    specialization: specialization,
+    name: name,
+    age: age,
+    parent_permission: true
+  }
+
+  if File.exist?('./data/people.json')
+    file = File.open('./data/people.json')
+
+    if File.empty?('./data/people.json')
+      teacher = [obj]
+    else
+      teacher = JSON.parse(File.read('./data/people.json'))
+      teacher << obj
+    end
+
+    file.close
+
+    myfile = File.open('./data/people.json', 'w')
+    myfile.write(JSON.pretty_generate(teacher))
+    myfile.close
+  end
+end
+
+def load_rentals
+  if File.exist?('./data/rentals.json')
+    file = File.open('./data/rentals.json')
+
+    if File.empty?('./data/rentals.json')
+      puts 'Please add some books first'
+    else
+      rentals = JSON.parse(File.read('./data/rentals.json'))
+      puts 'Reserved books: '
+      rentals.each do |rental|
+        puts "Name: #{rental['person']}, Book: #{rental['book']} on: #{rental['date']}"
+      end
+    end
+    file.close
+  else
+    puts 'No reservations found, please reseve some books first'
+  end
+end
```

Comment:
- [ ] I noticed that when I exit the application and restart the application, I can't get the rented book for a particular `ID`. This is because you don't store the `ID` of the person that the rental belongs to in the `JSON` files, so every time you exit and restart the application, the `ID` of the people that already exist changes. It would be nice to store the `ID` of the people in the JSON files and also access the rentals for a particular person based on the ID inputted by the user in the console application. 👍 💻 

![image](https://github.com/Roland-Ntwali/School-library-app/assets/77106962/258849e9-16a0-4c7f-b7b5-c8def69a0b44)